src/white.jl

```diff
@@ -0,0 +1,107 @@
+#
+# Copyright (C) 2020   Paul Söderlind
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export WhiteTest, BreuschPaganTest
+
+struct WhiteTest <: HypothesisTest
+    dof::Int                 #degrees of freedom in test
+    LM::Float64              #test statistic, distributed as Chisq(dof)
+    TestDescription::String  #short description of the test
+end
+
+"""
+    WhiteTest(X, e, TestType)
+
+Compute White's (or Breusch-Pagan's) test for heteroskedasticity.
+
+`X` is the matrix of regressors from the original model and `e` the vector of residuals.
+`TestType` is either `:Linear` for the Breusch-Pagan test, `:LinearAndSquares` for
+White's test with linear and squared terms only (no cross-products), or `:White` (the default)
+for the full White's test (linear, squared and cross-product terms). `X` should include a
+constant. The test statistic is T*R2 where R2 is from the regression of `e^2` on the terms
+mentioned above. Under the null hypothesis it is distributed as `Chisquare(dof)`
+where `dof` is the number of independent terms (not counting the constant), so the null
+is rejected when the test statistic is large enough.
+
+Implements: [`pvalue`](@ref)
+
+# References
+  * H. White, (1980): A heteroskedasticity-consistent covariance matrix estimator and a direct test for heteroskedasticity, Econometrica, 48, 817-838.
+  * T.S. Breusch & A.R. Pagan (1979), A simple test for heteroscedasticity and random coefficient variation, Econometrica, 47, 1287-1294
+
+# External links
+  * [White's test on Wikipedia](https://en.wikipedia.org/wiki/White_test)
+"""
+function WhiteTest(X::AbstractVecOrMat{<:Real}, e::AbstractVector{<:Real}, TestType=:White)
+    (n,K) = (size(X,1),size(X,2))    #nobs,nvars
+
+    n == length(e) || throw(DimensionMismatch("inputs must have the same length"))
+
+    if TestType == :Linear
+        z  = X
+        TestDescription = string("Breusch-Pagan's test for heteroskedasticity (linear)")
```

Comment:
There is no need to call `string` function here, string is any double-quoted text. Same below.

Reply:
Indeed. These are left-overs from another attempt. The fix will be in the next push.